src/main/java/com/github/steveice10/mc/protocol/packet/ingame/server/world/ServerAddVibrationSignalPacket.java

```diff
@@ -0,0 +1,42 @@
+package com.github.steveice10.mc.protocol.packet.ingame.server.world;
+
+import com.github.steveice10.mc.protocol.data.game.entity.metadata.Position;
+import com.github.steveice10.packetlib.io.NetInput;
+import com.github.steveice10.packetlib.io.NetOutput;
+import com.github.steveice10.packetlib.packet.Packet;
+import lombok.*;
+
+import java.io.IOException;
+
+//TODO
+@Data
+@With
+@Setter(AccessLevel.NONE)
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@AllArgsConstructor
+public class ServerAddVibrationSignalPacket implements Packet {
+    private Position origin;
+    private String identifier;
```

Comment:
I see this packet is still a WIP, but preliminarily, origin and identifier look like they should be NonNull.